runtime/interpreter/variable.go

```diff
@@ -19,11 +19,26 @@
 package interpreter
 
 type Variable struct {
-	Value Value
+	value  Value
+	getter func() Value
 }
 
-func NewVariable(value Value) *Variable {
+func (v *Variable) GetValue() Value {
+	if v.getter != nil {
+		v.value = v.getter()
+		v.getter = nil
+	}
+	return v.value
+}
+
+func (v *Variable) SetValue(value Value) {
+	v.getter = nil
+	v.value = value
+}
+
+func NewVariable(value Value, getter func() Value) *Variable {
```

Comment:
The signature implies it needs both the getter and the value (probably a personal taste :smile: ), but I guess it always needs only one of the two at a given time? Maybe having two constructors is better? (I miss overloading!)

Reply:
The alternative that comes to mind is to have a `func (v *Variable) WithValueGetter(getter func() Value) *Variable` receiver  instead of the extra constructor parameter (that is often null).

Reply:
Good idea! I'll add two different constructor functions, and yes, I do miss overloading, default arguments for parameters, etc. too :D

Reply:
Refactored in 1443910e43560dac8daf41d365df32b788e14f45